modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java

```diff
@@ -435,118 +404,70 @@ public void updateMessageID(String queueName, String nodeId, long startMessageId
      * @param nodeId node ID of the leaving node
      */
     public void reAssignSlotsWhenMemberLeaves(String nodeId) {
-        //Remove the entry from slot assignment map
 
-        HashmapStringTreeSetWrapper wrapper = slotAssignmentMap.remove(nodeId);
-        HashMap<String, TreeSet<Slot>> queueToSlotMap = null;
-        if (null != wrapper) {
-            queueToSlotMap = wrapper.getStringListHashMap();
-        }
-        if (queueToSlotMap != null) {
-            for (Map.Entry<String, TreeSet<Slot>> entry : queueToSlotMap.entrySet()) {
-                TreeSet<Slot> slotsToBeReAssigned = entry.getValue();
-                TreeSet<Slot> freeSlotTreeSet = new TreeSet<Slot>();
-                TreeSetSlotWrapper treeSetStringWrapper = new TreeSetSlotWrapper();
+        TreeSet<Slot> assignedSlotsSet = new TreeSet<Slot>();
```

Comment:
Maybe we don't need to create a new instance here ? the method below returns a new treeset ?